src/osyris/spatial/coordinate_transforms.py

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+from . import utils
+
+
+def translate(dataset, new_origin):
+    """
+    Translate all positionnal coordinates to new origin
+    """
+    for g in dataset.groups.keys():
+        for element in dataset[g]:
+            unit = dataset[g][element].unit
+            if unit.is_compatible_with("meter") and hasattr(dataset[g][element],
+                                                            "nvec"):
+                dataset[g][element] -= new_origin
+    dataset.origin = new_origin
+
+
+def rotate(dataset, new_basis, dr_L=None):
```

Comment:
> What is new_basis here? A single vector which is the normal?

Sort of. It is a single vector that describes the z axis of the new basis. This means that the original basis is simply rotated to align the old z axis with the new one (and hence avoids the need for us to provide the other two vectors that form the basis).

> Can we have the option to supply 3 vectors to fully define a basis also?

In that case I think we can simply project the old coordinates unto the new basis through a scalar product, instead of going through a rotation. But is this feature really necessary?

Reply:
> In that case I think we can simply project the old coordinates unto the new basis through a scalar product, instead of going through a rotation.

Then can't we just always do this? Without going through a rotation matrix? I think we need a rotation matrix if the user wants to specify an angle for rotation. But if they supply a normal vector, can't we just find two perpendicular vectors and make a new basis, and then use scalar products? This would make the code simpler?

> But is this feature really necessary?

Yes, this has been requested by some users.